### PR TITLE
[cleanup][tls] Unified management of JKS certificates

### PR DESCRIPTION
**Pull request by nodece**
Signed-off-by: Zixuan Liu <nodeces@gmail.com>

### Motivation

We have many JKS certificates in many modules, it is fragmented, and not conducive to our management.

### Modifications

- Use `tests/certificate-authority/jks` directory to manage the JKS certificates
- Copy  `tests/certificate-authority` to test target directory in each module that needs the tls certificates
- Modified and improve the load the JKS certificates
- Move the `generate_keystores_for_test.sh` to `tests/certificate-authority/generate_keystore.sh`

### Documentation

Need to update docs? 

- [x] `no-need-doc`

### CODE REVIEW

**Review by BewareMyPower — APPROVED** [review]

**Review by michaeljmarshall — APPROVED** [review]
LGTM, great addition @nodece.

Minor question: is `tests/certificate-authority/jks/cert-signed-client.pem` supposed to be an empty file?

**Comment by michaeljmarshall** [comment]
This PR includes doc updates, so I update the labels.

**Review by Shoothzj — APPROVED** [review]

**Comment by nodece** [comment]
> LGTM, great addition @nodece.
> 
> Minor question: is `tests/certificate-authority/jks/cert-signed-client.pem` supposed to be an empty file?

@michaeljmarshall Thanks for your report, it should be not empty.

**Comment by nodece** [comment]
It seems that don't trust the certificate. Do you have any good ways?
```
Failed reason: Received fatal alert: certificate_unknown
```

**Comment by nodece** [comment]
@michaeljmarshall @BewareMyPower @Shoothzj Generate the JKS script has been refactored, could you help review? Thanks for your time.

**Review by Shoothzj — APPROVED** [review]

**Review by mattisonchao — APPROVED** [review]

File: tests/certificate-authority/RemoveJksPassword.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+package main;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.security.KeyStore;
+
+// Main removes the password from the keystore or truststore
+public class Main {
```
Comment (Technoboy-) [inline tests/certificate-authority/RemoveJksPassword.java:27]: Do we need to name a better class name for this?

Comment (nodece) [reply tests/certificate-authority/RemoveJksPassword.java:27]: Use `RemoveJksPassword` instead of `Main`?

Comment (nodece) [reply tests/certificate-authority/RemoveJksPassword.java]: @Technoboy- Fixed.

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**